baybe/parameters/utils.py

```diff
@@ -87,3 +89,51 @@ def get_parameters_from_dataframe(
 def sort_parameters(parameters: Collection[Parameter]) -> tuple[Parameter, ...]:
     """Sort parameters alphabetically by their names."""
     return tuple(sorted(parameters, key=lambda p: p.name))
+
+
+def activate_parameter(
```

Comment:
@Scienfitz @AVHopp: This is one of the important open points to discuss. As you probably know by now, the mechanism of this PR is to divide parameters in to active and inactive sets. However, this opens a general question, not only related to this PR:
*  What do we even mean with  an "active continuous parameter", i.e. what is the value range where such a parameter is considered "active"? 

The current logic is to assume some small threshold around 0 to discriminate active/inactive ranges. And this function then takes care of "activating" a parameter according to this definition. 

*Problem to be discussed:*
For parameters with bound falling into the threshold range, everything is simple. But how can we guarantee that a parameter with bounds `(-a, b)` with `a,b >> threshold` is active? We cannot simply "cut out" the threshold interval since we then end up with a non-continuous range (see comment in docstring). This causes downstream problems in the sense that we cannot strictly guarantee that the "min_cardinality" part of a cardinality constrained will be fulfilled, since any optimizer might decide that – even if we assign the parameter to the "active set" – its value should be driven to zero. And then the cardinality will drop by one. In theory, one *could* solve this by branching into the two separate subspaces of the non-continuous range of the parameter, but that will end up with yet another exponential blow-up.

**TL;DR:** I think min-cardinality constraints are really hard to strictly fulfill. On the other hand, I cannot think of any use case where people actually need a min-constraint. So I see two options:
* (not preferred) dropping the option of min values in cardinality constraints. 
* (my suggestion) keeping everything as is and being aware (plus potentially mention it in the docstring) that the min-part of the constraint can be overridden by the optimizer. The min part still reduces the overall search burden in that it rules out certain subspaces to start with. But for the ones left, the min-part could be violated.

Input is appreciated!

Reply:
Quick follow-up: whatever we decide here, the corresponding logic of the random recommender should also be updated, which does not yet use any concept of activated parameters. It's not as important for the random recommender as it is for optimization based ones, but still ...

Reply:
First question: Wasn't this sort of a problem one of the main arguments against using Narrow Gaussian Processes? By my understanding, the problem that we are getting here is now basically identical, could you please give a quick heads-up why Narrow Gaussians are not an option/alternative? Don't need an extensive discussion, but rather a "Hey Alex, we already talked about that, let me quickly refresh your memory".

Reply:
Now to your actual points.

- First problem that I see with allowing `min_cardinality` and just mentioning in the docstring that this **might** be overwritten by the optimizer seems VERY dangerous to me. As we know by now, people tend to not read documentation **at all**. So what I see is people thinking "Oh, great, I can use a `min_cardinality`", stop to read and think at exactly that point, assume that it works as intended and then be angry if it does not work. 
- This is a potential crucial silent error that people might have at very inconvenient times: Imagine that you really require this lower cardinality, and that it works the first X runs of your optimization, just to fail in run X+1, invalidating potentially your whole set up.
- As you say yourself: "I cannot think of any use case where people actually need a min-constraint". We said that things like this should be a strong indicator to *not* implement stuff.

So I'd propose to drop the support for this.

Reply:
To first question: let's tackle this topic in a dedicated baybathon
Second: today in dev meeting

Reply:
As agreed in the baybathon, there are two options how to support the cardinality constraint (especially the minimum cardinality) in Botorch recommender: 

- Option I: this PR with hard threshold. If possible, the bounds of an active parameter gets updated by excluding the inactive range (-threshold, threshold). But if impossible, e.g. a parameter has bounds (-a, b) with a,b >> threshold (see discussion above), its value may be driven to zero by the optimizer and the `min_cardinality` might get violated. A warning `MinimumCardinalityViolatedWarning` will be raised if the minimum cardinality is violated.
- Option II: a future PR based on narrow Gaussian kernel = soft-constraint.